astropy/coordinates/sky_coordinate.py

```diff
@@ -501,6 +501,41 @@ def to_string(self, style='decimal', **kwargs):
 
         return coord_string
 
+    def is_equivalent_frame(self, other):
+        """
+        Checks if this object's frame as the same as that of the ``other``
+        object.
+
+        To be the same frame, two objects must be the same frame class and have
+        the same frame attributes. For two `SkyCoord` objects, *all* of the
+        frame attributes have to match, not just those relevant for the object's
+        frame.
+
+        Parameters
+        ----------
+        other : SkyCoord or BaseCoordinateFrame
+            The other object to check.
+
+        Returns
+        -------
+        isequiv : bool
+            True if the frames are the same, False if not.
+        """
+        if isinstance(other, BaseCoordinateFrame):
+            return self.frame.is_equivalent_frame(other)
+        elif hasattr(other, 'frame'):
```

Comment:
Is there a reason to not just test for `isinstance(other, SkyCoord)` here?  Duck typing is fine in principle, but really we're dealing with astropy coordinate classes or subclasses.  `SkyCoord` is a pretty specialized class and I can't imagine there being SkyCoord-ish objects that aren't subclasses.